File: _Src/Freem/Freem.Web/src/clients/models/records/RecordResponse.ts
export class RecordResponse {
  public readonly id: string;
  public readonly period: string;
  public readonly activities: [string];
  
  public readonly name?: string;
  public readonly description?: string;
  public readonly tags?: [string];

  constructor(
    id: string, 
    period: string, 
    activities: [string], 
    name?: string, 
    description?: string, 
    tags?: [string]
  ) {
    this.id = id;
    this.period = period;
    this.activities = activities;

    this.name = name;
    this.description = description;
    this.tags = tags;
  }
}

File: _Src/Freem/Freem.Web.Api.Public.Contracts/DTO/Records/RecordResponse.cs
using Freem.Entities.Abstractions.Relations.Collection;
using Freem.Entities.Activities.Identifiers;
using Freem.Entities.Records.Identifiers;
using Freem.Entities.Records.Models;
using Freem.Entities.Relations.Collections;
using Freem.Entities.Tags.Identifiers;
using Freem.Time.Models;

namespace Freem.Web.Api.Public.Contracts.DTO.Records;

public sealed class RecordResponse
{
    public RecordIdentifier Id { get; }
    
    public RecordName? Name { get; init; }
    public RecordDescription? Description { get; init; }
    
    public DateTimePeriod Period { get; }
    
    public IReadOnlyRelatedEntitiesIdentifiersCollection<ActivityIdentifier> Activities { get; }
    public IReadOnlyRelatedEntitiesIdentifiersCollection<TagIdentifier> Tags { get; }

    public RecordResponse(
        RecordIdentifier id,
        DateTimePeriod period,
        IReadOnlyRelatedEntitiesIdentifiersCollection<ActivityIdentifier> activities,
        IReadOnlyRelatedEntitiesIdentifiersCollection<TagIdentifier>? tags = null)
    {
        ArgumentNullException.ThrowIfNull(id);
        ArgumentNullException.ThrowIfNull(period);
        ArgumentNullException.ThrowIfNull(activities);
        
        Id = id;
        Period = period;
        Activities = activities;
        Tags = tags ?? RelatedTagsCollection.Empty;
    }
}